### PR TITLE
handwired/dactyl_manuform: add 6x7 version

### PR DESCRIPTION
**Pull request by matthew-dews**
## Description

This adds a 6x7 variant of the dactyl manuform. It is based on the 6x6 but with an additional column on each side.

## Types of Changes
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- I am assuming the 6x6 already matches the style guide
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/dactyl_manuform/6x7/6x7.c [hunk 1]
```diff
@@ -0,0 +1 @@
+#include "6x7.h"
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/6x7/6x7.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/handwired/dactyl_manuform/6x7/6x7.h [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#pragma once
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/6x7/6x7.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/handwired/dactyl_manuform/6x7/keymaps/default/config.h [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/6x7/keymaps/default/config.h:18]: All of these are the defaults.  This file can be removed.

File: keyboards/handwired/dactyl_manuform/6x7/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,46 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/6x7/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/handwired/dactyl_manuform/6x7/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+#define _QWERTY 0
+#define _LOWER 1
+#define _RAISE 2
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/6x7/keymaps/default/keymap.c:5]:
```suggestion
enum custom_layer {
    _QWERTY,
    _LOWER,
    _RAISE,
};
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/dactyl_manuform/6x7/keymaps/default/config.h (hunk 3 above)
Comment (matthew-dews) [reply keyboards/handwired/dactyl_manuform/6x7/keymaps/default/config.h:18]: Even the matrix pins? Where are the defaults defined?

Comment (matthew-dews) [reply keyboards/handwired/dactyl_manuform/6x7/keymaps/default/config.h:18]: Sorry, looked at wrong file

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**